File: busy_man/busy_man.py
test_cases = int(input())
for _ in range(test_cases):
    activities = int(input())
    pairs = []
    for _ in range(activities):
        data = input().split(" ")
        pairs.append((int(data[0]), int(data[1])))

    pairs.sort(key=lambda x: x[1])

    if(len(pairs) > 0):
        solution = [pairs.pop(0)]
        for element in pairs:
            if solution[-1][1] <= element[0]:
                solution.append(element)

    print(len(solution))

File: busy_man/busy_man.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int test_cases;
    cin >> test_cases;

    for (int t = 0; t < test_cases; t++) {
        int activities;
        cin >> activities;

        vector<pair<int, int> > pairs;
        for (int i = 0; i < activities; i++) {
            int start, end;
            cin >> start >> end;
            pairs.push_back(make_pair(start, end));
        }

        sort(pairs.begin(), pairs.end(), [](const pair<int, int> &a, const pair<int, int> &b) {
            return a.second < b.second;
        });

        vector<pair<int, int> > solution;
        if(pairs.size() > 0) {
            solution.push_back(pairs[0]);

            for (size_t i = 1; i < pairs.size(); i++) {
                if (solution.back().second <= pairs[i].first) {
                    solution.push_back(pairs[i]);
                }
            }
        }

        cout << solution.size();
        cout << endl;
    }

    return 0;
}
